src/io/geotiff.rs

```diff
@@ -18,6 +20,55 @@ impl<R: Read + Seek> CogReader<R> {
 
         Ok(Self { decoder })
     }
+
+    /// Affine transformation for 2D matrix extracted from TIFF tag metadata, used to transform
+    /// image pixel (row, col) coordinates to and from geographic/projected (x, y) coordinates.
+    ///
+    /// ```text
+    /// | x' |   | a b c | | x |
+    /// | y' | = | d e f | | y |
+    /// | 1  |   | 0 0 1 | | 1 |
+    /// ```
+    ///
+    /// where (`x'` and `y'`) are world coordinates, and (`x`, `y) are the pixel's image
+    /// coordinates. Letters a to f represent:
+    ///
+    /// - `a` - width of a pixel (x-resolution)
+    /// - `b` - row rotation (typically zero)
+    /// - `c` - x-coordinate of the *center* of the upper-left pixel (x-origin)
+    /// - `d` - column rotation (typically zero)
+    /// - `e` - height of a pixel (y-resolution, typically negative)
+    /// - `f` - y-coordinate of the *center* of the upper-left pixel (y-origin)
+    ///
+    /// References:
+    /// - <https://docs.ogc.org/is/19-008r4/19-008r4.html#_coordinate_transformations>
+    fn transform(&mut self) -> TiffResult<AffineTransform<f64>> {
+        // Get x and y axis rotation (not yet implemented)
+        let (x_rotation, y_rotation): (f64, f64) =
+            match self.decoder.get_tag_f64_vec(Tag::ModelTransformationTag) {
+                Ok(_model_transformation) => unimplemented!("Non-zero rotation is not handled yet"),
+                Err(_) => (0.0, 0.0),
+            };
```

Comment:
Note to future self (or someone else), find a GeoTIFF with non-zero rotation values and implement this properly.